### ISSUE TITLE
readFormData() doesn't work

### ISSUE THREAD

**Issue by luca-smartpricing** [unjs/h3#561]
### Environment

- Operating System: Darwin
- Node Version:     v18.18.2
- Nuxt Version:     3.7.4
- CLI Version:      3.9.0
- Nitro Version:    2.6.3
- Package Manager:  yarn@3.5.0
- Builder:          -
- User Config:      telemetry, vite, nitro, experimental, app, runtimeConfig, css, modules, site, sitemap, googleFonts, tailwindcss, prismic, i18n, devtools
- Runtime Modules:  @nuxt/devtools@0.8.5, @nuxtjs/google-fonts@3.0.2, @nuxtjs/tailwindcss@6.8.0, @vueuse/nuxt@10.4.1, @nuxtjs/prismic@3.0.3, @nuxtjs/i18n@8.0.0-rc.4, nuxt-simple-sitemap@3.3.4, nuxt-simple-robots@3.1.6
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-vcth9o?file=app.vue

### Describe the bug

Hi! 
When i try to read FormData with `const formData = await readFormData(event)` I receive this error: 
`[nuxt] [request error] [unhandled] [500] Error: Unexpected end of multipart data`
It's come from Busboy

The problem appear when i set `Content-Type` on nuxt `useFetch` (also if i set lowercase `content-type`)

```js
const headers = new Headers({
    'Content-Type': 'multipart/form-data; boundary=someBoundary',
  })
  const formData = new FormData()
  formData.append('test', 'string')

  const { data, error } = await useFetch('/api/file-upload', {
    method: 'POST',
    headers,
    body: formData,
  })
```

If i don't set `Content-Type` in local dev work perfect but on Netlify the problem appear again. 
Did you know why? 
Thank guys!

**Comment by PatSosna** [comment]
This doesn't work for me either. You can use workaround for this readMultipartFormData, you can find it here https://github.com/unjs/nitro/issues/1721

**Comment by passionate-bram** [comment]
I noticed that, on netlify, `await readRawBody(event)` returns the formdata encoded as base64. While locally it returns the form data as-is (not encoded into base64).

**Comment by Atinux** [comment]
Thanks for your help @passionate-bram 

I think the main issue here is that `readFormData` returns a stream for the `body` which is very useful when uploading files for instance (https://github.com/unjs/h3/blob/e38d4446354aba7e28d0971737f31787f25c4cad/src/utils/request.ts#L189).

**Comment by Atinux** [comment]
Actually could you try with `h3-nightly`?

in your `package.json` (if using `pnpm`):

```json
"resolutions": {
  "h3": "npm:h3-nightly@latest"
}
```

**Comment by passionate-bram** [comment]
> Actually could you try with `h3` [v1.10.1](https://github.com/unjs/h3/releases/tag/v1.10.1).
> 
> Seems the issue has been fixed in #616

Yes, I noticed that as well after reviewing the nitro preset code for netlify/aws. I have upgraded my nuxt to 3.9 which ends up updating h3 to include the base64 body fix.
